src/models/message_status.ts

```diff
@@ -22,11 +22,18 @@ export const MESSAGE_STATUS_MODEL_PK_FIELD = "messageId" as const;
 
 // We cannot intersect with MessageStatus
 // as it is a *strict* interface
-export const MessageStatus = t.interface({
-  messageId: NonEmptyString,
-  status: MessageStatusValue,
-  updatedAt: Timestamp
-});
+export const MessageStatus = t.intersection([
+  t.interface({
+    messageId: NonEmptyString,
+    status: MessageStatusValue,
+    updatedAt: Timestamp
+  }),
+  t.partial({
+    isRead: t.boolean,
+    // eslint-disable-next-line sort-keys
+    isArchived: t.boolean
+  })
```

Comment:
What if we decide to let these props required with a default ?

Reply:
You're right, I'll try this way

Reply:
This a good point. By default a message is not read and not archived.

Reply:
We have to check the case that a messageStatus is stored without these props and we perform a retrieve operation that trigger a decode :)

Reply:
Yes.. I actually don't like very much the idea of storing the two values in the initial steps, when "status" gets changed, but it's more clear when retrieving them

Reply:
Done in [0b73fe3](https://github.com/pagopa/io-functions-commons/pull/239/commits/0b73fe318cd16416a6772dd52b2623d480e7812e)
Test updated in [670d905](https://github.com/pagopa/io-functions-commons/pull/239/commits/670d9058d57d0f51598b643d78db13f20319c69a)